functions/src/modules/SQR.functions.ts

```diff
@@ -370,3 +384,171 @@ export const importSpreadSheetData = functions.https.onRequest(
+ */
+export interface IAllotmentRow {
+  // some of these should ideally be enums
+  days_passed: string;
+  date_given: string;
+  notes: string;
+  language: string;
+  status: string;
+  file_name: string;
+  devotee: string;
+  email: string;
+  phone: string;
+  location: string;
+  date_done: string;
+  follow_up: string;
+  list: string;
+  serial: number;
+}
+
+export interface ISubmissionRow {
+  // some of these should ideally be enums
+  completed: string;
+  updated: string;
+  submission_serial: number;
+  update_link: string;
+  audio_file_name: string;
+  unwanted_parts: string;
+  sound_issues: string;
+  sound_quality_rating: string;
+  beginning: string;
+  ending: string;
+  comments: string;
+  name: string;
+  email_address: string;
+}
+
+/**
+ * On creation of a new allotment record id, update and sync data values to Google Spreadsheets
+ *
+ */
+export const syncAllotments = functions.database
+  .ref('/files/{listName}/{fileName}')
+  .onUpdate(
+    async (
+      change: functions.Change<functions.database.DataSnapshot>,
+      context: functions.EventContext
+    ): Promise<any> => {
+      const { listName, fileName } = context.params;
+      const changedValues = change.after.val();
+      const spreadsheetId = functions.config().sqr.spreadsheet_id;
+
+      const gsheets = new GoogleSheet();
+      const allotmentFileNames = await gsheets.getColumn(
+        spreadsheetId,
+        ISoundQualityReportSheet.Allotments,
+        'File Name'
+      );
+
+      const rowNumber = allotmentFileNames.indexOf(fileName) + 1;
+      const { languages, notes, soundQualityReporting } = changedValues;
+      const {
+        timestampGiven,
+        assignee,
+        timestampDone,
+        followUp,
+      } = soundQualityReporting;
+      const updateResults = await gsheets.updateAllotmentRow(
+        spreadsheetId,
+        ISoundQualityReportSheet.Allotments,
+        rowNumber,
+        {
+          days_passed: EMPTY_VALUE,
+          date_given: spreadsheetDateFormat(timestampGiven),
+          notes: withDefault(notes),
+          language: commaSeparated(languages),
+          status: soundQualityReporting.status,
+          file_name: fileName,
+          devotee: withDefault(assignee.name),
+          email: withDefault(assignee.emailAddress),
+          phone: EMPTY_VALUE,
+          location: EMPTY_VALUE,
+          date_done: spreadsheetDateFormat(timestampDone),
+          follow_up: withDefault(followUp),
+          list: withDefault(listName),
+          serial: EMPTY_VALUE,
+        }
+      );
+      console.log('Update results: ', updateResults.data);
+    }
+  );
+
+
+interface IAudioDescription {
+  beginning: string; // h:mm:ss
+  ending: string; // h:mm:ss
+  type: string;
+  description: string;
+}
+
+/**
+ * Used for Unwanted Parts and Sound Issues to create multi-line comments
+ *
+ */
+export function formatMultilineComment(audioDescriptionList: IAudioDescription[]) {
```

Comment:
Can't we add a method to the `IAudioDescription` interface (or a derived class) to format a single description, and then join them with `\n`?

Reply:
Yes, something like that.